service/register.go

```diff
@@ -13,37 +13,55 @@ const (
 	HEARTBEAT_DURATION = 5
 )
 
-func Register(service string, host *Host, stop chan struct{}) (chan struct{}, error) {
+type Infos struct {
+	Critical bool `json:"critical"` // Is the service critical to the infrastructure health?
+}
+
+func Register(service string, host *Host, infos *Infos, stop chan struct{}) (chan struct{}, error) {
 	if len(host.Name) == 0 {
 		host.Name = hostname
 	}
 
 	registered := make(chan struct{}, 1)
-	key := fmt.Sprintf("/services/%s/%s", service, host.Name)
+	hostKey := fmt.Sprintf("/services/%s/%s", service, host.Name)
 	hostJson, _ := json.Marshal(&host)
-	value := string(hostJson)
+	hostValue := string(hostJson)
+
+	serviceKey := fmt.Sprintf("/services_infos/%s", service)
+	serviceJson, _ := json.Marshal(infos)
+	serviceValue := string(serviceJson)
 
 	go func() {
 		ticker := time.NewTicker((HEARTBEAT_DURATION - 1) * time.Second)
-		KAPI().Set(context.Background(), key, value, &etcd.SetOptions{TTL: HEARTBEAT_DURATION * time.Second})
+		KAPI().Set(context.Background(), hostKey, hostValue, &etcd.SetOptions{TTL: HEARTBEAT_DURATION * time.Second})
+
+		if infos != nil {
+			KAPI().Set(context.Background(), serviceKey, serviceValue, nil)
```

Comment:
That's just that it make less sense. An host is'nt critical, the service is. And if a service has no node, how do we know if the service is critical for the infrastructure (this could be useful for the `Philae` project)